### PR TITLE
UF-258: Plugins can be copied or renamed to non-editable plugin names

### PR DESCRIPTION
**Pull request by paulovmr**
- Inserted validation of duplicated name on plugins;

- Separated the view and the presenter code on the NewPluginPopUp;

- Introduced tests to the file name validation.

Depends on: https://github.com/uberfire/uberfire/pull/267

### CODE REVIEW

**Comment by paulovmr** [comment]
PR revised by Hangouts with @ederign. I will do some modifications on it.

**Comment by paulovmr** [comment]
@ederign Ready for another review. :)

**Comment by ederign** [comment]
@paulovmr nice MVP extract.

Could you please add PluginsInfoTest, NameValidatorTest and RuleValidatorTest?

With these tests, we are good to go.

File: uberfire-runtime-plugins/uberfire-runtime-plugins-client/src/main/java/org/uberfire/ext/plugin/client/info/PluginsInfo.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    private SplashPluginResourceType splashPluginResourceType;
+
+    private DynamicMenuResourceType dynamicMenuResourceType;
+
+    private ActivityBeansInfo activityBeansInfo;
+
+    private ClientTypeRegistry clientTypeRegistry;
+
+    protected PluginsInfo() {
```
Comment (ederign) [inline uberfire-runtime-plugins/uberfire-runtime-plugins-client/src/main/java/org/uberfire/ext/plugin/client/info/PluginsInfo.java]: I'm not 100% sure, but I think tomcat requires public constructors. Please change it.

**Comment by ederign** [comment]
@paulovmr please fix the constructor and we have a +1.

**Comment by paulovmr** [comment]
@ederign fixed.

**Comment by csadilek** [comment]
master: 45919d49e9d3eca9c5b70dab3da460a905362213

**Pull request closed without merging**